lib/resty/core/base.lua

```diff
@@ -174,6 +209,20 @@ function _M.ref_in_table(tb, key)
 end
 
 
+function _M.check_subsystem(...)
+    local subsystems = {...}
+    local total = select("#", ...)
+
+    for i = 1, total do
+        if subsystems[i] == subsystem then
+            return
+        end
+    end
+
+    error("unsupported subsystem: " .. subsystem)
```

Comment:
Seems like there is no test cases triggering this error?

Reply:
Added for both `stream` and `http` subsystem.